Framework/Beamline/inc/MantidBeamline/DetectorInfo.h

```diff
@@ -68,41 +68,87 @@ class MANTID_BEAMLINE_DLL DetectorInfo {
   bool isEquivalent(const DetectorInfo &other) const;
 
   size_t size() const;
+  bool isScanning() const;
 
   bool isMonitor(const size_t index) const;
+  bool isMonitor(const std::pair<size_t, size_t> index) const;
   bool isMasked(const size_t index) const;
+  bool isMasked(const std::pair<size_t, size_t> index) const;
   void setMasked(const size_t index, bool masked);
+  void setMasked(const std::pair<size_t, size_t> index, bool masked);
   Eigen::Vector3d position(const size_t index) const;
+  Eigen::Vector3d position(const std::pair<size_t, size_t> index) const;
   Eigen::Quaterniond rotation(const size_t index) const;
+  Eigen::Quaterniond rotation(const std::pair<size_t, size_t> index) const;
   void setPosition(const size_t index, const Eigen::Vector3d &position);
+  void setPosition(const std::pair<size_t, size_t> index,
+                   const Eigen::Vector3d &position);
   void setRotation(const size_t index, const Eigen::Quaterniond &rotation);
+  void setRotation(const std::pair<size_t, size_t> index,
+                   const Eigen::Quaterniond &rotation);
+
+  size_t scanCount(const size_t index) const;
+  std::pair<int64_t, int64_t>
+  scanInterval(const std::pair<size_t, size_t> index) const;
+  void setScanInterval(const std::pair<size_t, size_t> index,
+                       std::pair<int64_t, int64_t> interval);
+
+  void merge(const DetectorInfo &other);
 
 private:
+  size_t linearIndex(const std::pair<size_t, size_t> &index) const;
+  void checkNoTimeDependence() const;
+  void initScanCounts();
+  void initScanIntervals();
+  void initIndices();
+
   Kernel::cow_ptr<std::vector<bool>> m_isMonitor{nullptr};
   Kernel::cow_ptr<std::vector<bool>> m_isMasked{nullptr};
   Kernel::cow_ptr<std::vector<Eigen::Vector3d>> m_positions{nullptr};
   Kernel::cow_ptr<std::vector<Eigen::Quaterniond>> m_rotations{nullptr};
+
+  Kernel::cow_ptr<std::vector<size_t>> m_scanCounts{nullptr};
+  Kernel::cow_ptr<std::vector<std::pair<int64_t, int64_t>>> m_scanIntervals{
```

Comment:
Some comment about using the same underlying type as `Kernel::DataAndTime` is needed here.